dsgrid/query/query_submitter.py

```diff
@@ -115,6 +116,16 @@ def _make_table_handler(self, context: QueryContext):
             raise NotImplementedError(f"Unsupported {context.get_table_format_type()=}")
         return handler
 
+    def _run_checks(self, model):
+        subsets = set(self.project.config.list_dimension_query_names(DimensionCategory.SUBSET))
+        for agg in model.result.aggregations:
+            for _, column in agg.iter_dimensions_to_keep():
+                dimension_query_name = column.dimension_query_name
+                if column.dimension_query_name in subsets:
+                    raise DSGInvalidQuery(
+                        f"subset dimensions cannot be used in aggregations: {dimension_query_name=}"
```

Comment:
Is it difficult to give full set dimension name of the subset dimension as hint?

Reply:
It is not difficult to do that. But what if they confused a subset dimensions with a supplemental? I could print base + all supplemental. Or the user could go back to the CLI or GUI and look themselves.

Reply:
Aggregation only takes base or supplemental base right, so there's a finite options. You can hint and be like "do you mean <this base dimension>? 

If they are confused about what they are doing, then there's a bigger issue on their part.

Reply:
To be clear, supplemental dimensions can also be used for aggregations, and so I would have to list them all out in a well-formatted way.